workspacehelper/k8s_secret.go

```diff
@@ -0,0 +1,69 @@
+package workspacehelper
+
+import (
+	"context"
+	"errors"
+
+	"github.com/hashicorp/terraform-k8s/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// GetSecretData retrieves the data from a secret in a given namespace
+func (r *WorkspaceHelper) GetSecretData(namespace string, name string) (map[string][]byte, error) {
+	// If no secretName defined, return empty map
+	if name == "" {
+		return make(map[string][]byte), nil
```

Comment:
Thanks for reviewing my changes, @arybolovlev. I did this logic because [another PR](https://github.com/hashicorp/terraform-k8s/pull/133/files) is [using](https://github.com/hashicorp/terraform-k8s/pull/133/files#diff-7db3618f5a31e3b525625d1cfc62e970bae9e418121877c10cd8f9d851e17767R298) the same method. The `instance.Spec.SecretName` can be undefined which leads to "empty string" which is not an error in that context.

Reply:
I would say that the verification for the empty secret name should be done before calling this function if you do a schema attribute validation and you don't want to get an error message. In this case, it will be more explicit behavior and more obvious code. The `GetSecretData` function sounds generic enough and the expectation from it is to get an error message if the secret can't be found, which is the case for the empty secret name.